apps/blockchain/test/blockchain/backend_test.exs

```diff
@@ -0,0 +1,134 @@
+# Copyright 2019 OmiseGO Pte Ltd
+#
+# Licensed under the Apache License, Version 2.0 (the "License");
+# you may not use this file except in compliance with the License.
+# You may obtain a copy of the License at
+#
+#     http://www.apache.org/licenses/LICENSE-2.0
+#
+# Unless required by applicable law or agreed to in writing, software
+# distributed under the License is distributed on an "AS IS" BASIS,
+# WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+# See the License for the specific language governing permissions and
+# limitations under the License.
+
+defmodule Blockchain.BackendTest do
+  use ExUnit.Case
+  alias Blockchain.Backend
+  alias Ecto.UUID
+
+  defmodule DumbBackend do
+    def start_link, do: GenServer.start_link(__MODULE__, :ok, [])
+    def init(:ok), do: {:ok, nil}
+    def stop(pid), do: GenServer.stop(pid)
+
+    def handle_call(:generate_wallet, _from, reg) do
+      {:reply, {:ok, "wallet_id", "public_key"}, reg}
+    end
+  end
+
+  setup do
+    mock_id = UUID.generate()
+    mock_key = UUID.generate()
+
+    Code.eval_string("""
```

Comment:
Curious question: Why it needs to be wrapped around `Code.eval_string`?

Reply:
Because we’re dynamically returning the value of `UUID.generate()` above (which also need to be passed as state for comparison). 

Unfortunately it seems like the variable above are in function scope and cannot be accessed inside `defmodule`. AFAIK Elixir also use the same pattern quite a lot in their test case so it should be safe to use.